File: ReadyChef.Core/Models/RecipeIngredient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ReadyChef.Core.Models
{
    public class RecipeIngredient
    {
        public Ingredient Ingredient { get; set; }
        public int Amount { get; set; }
        public string AmountUnits { get; set; }
    }
}

File: ReadyChef.Api/Scripts/services/recipeIngredient.js
angular.module('readyChefApp')
    .service('recipeIngredient', recipeIngredient);
recipeIngredient.$inject = [];
function recipeIngredient() {
    function newRecipeIngredient(ingredient, amount, amountUnits) {
        this.ingredient = ingredient;
        this.amount = amount;
        this.amountUnits = amountUnits;
    };
    return newRecipeIngredient;
};